.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,33 @@
+name: Integration Tests
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
I think we possibly could run on this on PR so long as it's not a build the blocks merge.

Reply:
In addition to running on PR and master builds, should we run on release branch builds as well?

Reply:
I'm fine running or not running in release branch builds.
As long as every commit to release branches is done through PRs, the tests will run in those PRs checks before the merge. 
Moreover, not running in release branches will make that the full test suite is executed at least once in each PR regardless of any code change, because there will be no cache to inherit from. 

In the case of `master` its PRs will inherit the cache that was saved in `master` runs, at least only on the first commit in the PR. The logic is literally "get the cache for this commit sha, if not found, get the cache from the previous commit sha (if exists)".